### PR TITLE
8287738: [PPC64] jdk/incubator/vector/*VectorTests failing

### PR DESCRIPTION
**Pull request by TheRealMDoerr**
`PopCountVI` needs to handle several types after [JDK-8284960](https://bugs.openjdk.java.net/browse/JDK-8284960).
(See https://github.com/openjdk/jdk/commit/6f6486e97743eadfb20b4175e1b4b2b05b59a17a.)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287738](https://bugs.openjdk.org/browse/JDK-8287738): [PPC64] jdk/incubator/vector/*VectorTests failing

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [e439bf3d](https://git.openjdk.java.net/jdk/pull/8998/files/e439bf3daec60b3b06ced4ab6ae33d8cb8c9054c)
 * [Goetz Lindenmaier](https://openjdk.java.net/census#goetz) (@GoeLin - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8998/head:pull/8998` \
`$ git checkout pull/8998`

Update a local copy of the PR: \
`$ git checkout pull/8998` \
`$ git pull https://git.openjdk.java.net/jdk pull/8998/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8998`

View PR using the GUI difftool: \
`$ git pr show -t 8998`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8998.diff">https://git.openjdk.java.net/jdk/pull/8998.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
I am not expert of PPC64 but changes look good to me.

File: src/hotspot/cpu/ppc/ppc.ad [hunk 1]
```diff
@@ -14098,13 +14098,28 @@ instruct vsqrt2D_reg(vecX dst, vecX src) %{
...
+      case T_BYTE:
+        __ vpopcntb($dst$$VectorSRegister->to_vr(), $src$$VectorSRegister->to_vr());
+        break;
+      case T_SHORT:
+        __ vpopcnth($dst$$VectorSRegister->to_vr(), $src$$VectorSRegister->to_vr());
+        break;
+      case T_INT:
+        __ vpopcntw($dst$$VectorSRegister->to_vr(), $src$$VectorSRegister->to_vr());
+        break;
+      case T_DOUBLE:
```
Comment (GoeLin) [inline src/hotspot/cpu/ppc/ppc.ad:14117]: I think this should be T_LONG.

**Review by GoeLin — APPROVED** [review]
LGTM

**Comment by TheRealMDoerr** [comment]
Thanks for the reviews!
/integrate

**Pull request closed without merging**